addons/account/tests/test_chart_template.py

```diff
@@ -212,3 +212,42 @@ def test_update_taxes_multi_company(self):
         ])
         # we should have 4 records: 2 companies * (1 original tax + 1 recreated tax)
         self.assertEqual(len(taxes_from_template_1), 4)
+
+    def test_message_to_accountants(self):
+        """ When we duplicate a tax because it was too different from the existing one we send
+        a message to accountant advisors. This message should only be sent to advisors
+        and not to regular users.
+        """
+        # create 1 normal user, 2 accountants managers
+        accountant_manager_group = self.env.ref('account.group_account_manager')
+        advisor_users = self.env['res.users'].create([{
+            'name': 'AccountAdvisorTest1',
+            'login': 'aat1',
+            'password': 'aat1aat1',
+            'groups_id': [(4, accountant_manager_group.id)],
+        }, {
+            'name': 'AccountAdvisorTest2',
+            'login': 'aat2',
+            'password': 'aat2aat2',
+            'groups_id': [(4, accountant_manager_group.id)],
+        }])
+        normal_user = self.env['res.users'].create([{
+            'name': 'AccountUserTest1',
+            'login': 'aut1',
+            'password': 'aut1aut1',
+            'groups_id': [(4, self.env.ref('account.group_account_user').id)],
+        }])
+        # create situation where we need to recreate the tax during update to get notification(s) sent
+        self.tax_template_1.amount += 1
+        update_taxes_from_templates(self.env.cr, self.chart_template_xmlid)
+
+        # accountants received the message
+        self.assertEqual(self.env['mail.message'].search_count([
+            ('partner_ids', 'in', advisor_users.partner_id.ids),
+            ('body', 'like', f"%{self.tax_template_1.name}%"),
+        ]), 1)
+        # normal user didn't
+        self.assertEqual(self.env['mail.message'].search_count([
+            ('partner_ids', 'in', normal_user.partner_id.ids),
+            ('body', 'like', f"%{self.tax_template_1.name}%"),
```

Comment:
I agree your solution is more explicit but it is less robust to change since we will have to hardcode the subject in tests. Thus the test will break if the subject content changes, which is it suppose to be a meaningless change for what we are testing.
While with a specific-enough tax template name, specifically created for this test the chance to catch the wrong message is almost non-existent... but it doesn't help for explicitness :)